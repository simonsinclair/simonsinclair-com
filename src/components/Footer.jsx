import PropTypes from 'prop-types';

const Footer = ({ date }) => (
  <footer className="container">
    <p>
      <small>
        &copy;
        {' '}
        {date.getFullYear()}
        {' '}
        Simon Sinclair.
      </small>
    </p>

    <style jsx>
      {`
        footer {}
      `}
    </style>
  </footer>
);

Footer.propTypes = {
  date: PropTypes.instanceOf(Date).isRequired,
};

export default Footer;
