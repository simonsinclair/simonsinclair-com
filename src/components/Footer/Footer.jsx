import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledFooter = styled.footer`
  margin-top: 3rem;
  padding-bottom: 1.5rem;
`;

const Footer = ({ date }) => (
  <StyledFooter className="container">
    <p>
      <small>
        &copy;
        {' '}
        {date.getFullYear()}
        {' '}
        Simon Sinclair.
      </small>
    </p>
  </StyledFooter>
);

Footer.propTypes = {
  date: PropTypes.instanceOf(Date).isRequired,
};

export default Footer;
