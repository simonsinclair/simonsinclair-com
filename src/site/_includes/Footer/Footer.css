.footer {
  margin-top: 3rem;
  padding-bottom: 1.5rem;
}
