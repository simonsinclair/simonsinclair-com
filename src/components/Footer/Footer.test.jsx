import { render } from '@testing-library/react';

import Footer from './Footer';

describe('<Footer /> is a component that', () => {
  it('renders and matches its snapshot', () => {
    const { container } = render(<Footer date={new Date(88, 0, 15)} />);
    expect(container).toMatchSnapshot();
  });

  it('renders the given copyright year', () => {
    const { queryByText } = render(<Footer date={new Date(88, 0, 15)} />);
    expect(queryByText(/1988/i)).not.toBeNull();
  });
});
