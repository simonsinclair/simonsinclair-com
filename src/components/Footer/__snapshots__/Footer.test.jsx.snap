// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> is a component that renders and matches its snapshot 1`] = `
.c0 {
  margin-top: 3rem;
  padding-bottom: 1.5rem;
}

<div>
  <footer
    class="container c0"
  >
    <p>
      <small>
        Â©
         
        1988
         
        Simon Sinclair.
      </small>
    </p>
  </footer>
</div>
`;
